### PR TITLE
implement modplug music module

### PR DESCRIPTION
**Pull request by rfomin**
`vcpkg` only has a very old `libxmp-lite` variant, and `openmpt` is overly complex, so I think modplug is good enough.

### CODE REVIEW

File: src/i_modmusic.c [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+static ModPlug_Settings settings;
+
+static uint32_t MOD_Callback(byte *buffer, uint32_t buffer_samples)
+{
+    int filled;
+
+    filled = ModPlug_Read(modplug.file, buffer, buffer_samples * 4);
+
+    return filled / 4;
```
Comment (fabiangreffrath) [inline src/i_modmusic.c:43]: Is this another 8-bit mono / 16-bit stereo calculation?

Comment (rfomin) [reply src/i_modmusic.c:43]: It's 16 bit stereo.

File: src/i_modmusic.c [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+}
+
+static void *I_MOD_RegisterSong(void *data, int len)
+{
+    ModPlugFile *file;
+
+    modplug.data = data;
+    modplug.size = len;
+
+    file = ModPlug_Load(data, len);
```
Comment (fabiangreffrath) [inline src/i_modmusic.c:91]: This is just to check if data contains a valid tracker signature, right?

File: src/i_sound.c [hunk 3]
```diff
@@ -51,6 +52,14 @@ static music_modules_t music_modules[] =
     { &music_opl_module, 1 },
 };
 
+static music_module_t *persistent_modules[] =
```
Comment (fabiangreffrath) [inline src/i_sound.c:55]: What does "persistent" mean in this context?

File: src/i_modmusic.c (hunk 2 above)
Comment (rfomin) [reply src/i_modmusic.c:91]: Yes. It's all a little awkward, I plan to try a different approach.

File: src/i_sound.c [hunk 4]
```diff
@@ -717,22 +735,30 @@ boolean IsMus(byte *mem, int len)
...
-        if (midi_player_module)
-        {
-            active_module = midi_player_module;
-            return active_module->I_RegisterSong(data, size);
-        }
+        active_module = midi_player_module;
+        return active_module->I_RegisterSong(data, size);
     }
 
     if (midi_player_module == &music_opl_module)
```
Comment (fabiangreffrath) [inline src/i_sound.c:746]: I always choke on this. Could you add a comment that this is the non-MIDI code path? It's easy to miss the `return` statement in the condition above.

File: src/i_sound.c (hunk 3 above)
Comment (rfomin) [reply src/i_sound.c:55]: These modules are always loaded, so we can test many music formats on every song change.

Comment (fabiangreffrath) [reply src/i_sound.c:55]: Good idea, could you please put this into a comment?

File: src/i_modmusic.c (hunk 2 above)
Comment (rfomin) [reply src/i_modmusic.c:91]: `libxmp` can be difficult to configure on all platforms. I'll try it too.

File: src/i_sound.c (hunk 4 above)
Comment (rfomin) [reply src/i_sound.c:730]: Done in https://github.com/fabiangreffrath/woof/pull/976/commits/fa635ac4161ef89d2528b251bada8eebd3c3d9c9

File: src/i_sound.c (hunk 3 above)
Comment (rfomin) [reply src/i_sound.c]: I removed `persistent_modules`, now `music_oal_module` acts like SDL_Mixer before.

Comment (fabiangreffrath) [reply src/i_sound.c]: Looks very clean now!

**Review by fabiangreffrath — APPROVED** [review]

**Pull request merged**